pkg/sfu/downtrack.go

```diff
@@ -854,76 +857,100 @@ func (d *DownTrack) CreateSenderReport() *rtcp.SenderReport {
 	return d.rtpStats.GetRtcpSenderReport(d.ssrc)
 }
 
-func (d *DownTrack) writeBlankFrameRTP(duration float32) error {
-	// don't send if nothing has been sent
-	if !d.rtpStats.IsActive() {
-		return nil
-	}
+func (d *DownTrack) writeBlankFrameRTP(duration float32, generation uint32) chan struct{} {
+	done := make(chan struct{})
+	go func() {
+		// don't send if nothing has been sent
+		if !d.rtpStats.IsActive() {
+			close(done)
+			return
+		}
 
-	// LK-TODO: Support other audio/video codecs
-	if d.mime != "audio/opus" && d.mime != "video/vp8" && d.mime != "video/h264" {
-		return nil
-	}
+		// LK-TODO: Support other audio/video codecs
+		if d.mime != "audio/opus" && d.mime != "video/vp8" && d.mime != "video/h264" {
```

Comment:
@boks1971 like this

```suggestion
		var writeBlankFrame func(*rtp.Header, bool)
switch d.mime {
case "audio/opus":
writeBlankFrame = d.writeOpusBlankFrame
case "video/vp8":
writeBlankFrame = d.writeVP8BlankFrame
case "video/h264":
writeBlankFrame = d.writeH264BlankFrame
default:
close(done)
return
```

Reply:
Got it. Thank you @cnderrauber . Will make the change.

Reply:
Added in this commit @cnderrauber - https://github.com/livekit/livekit/pull/710/commits/01c67a19388e9f6da73229683e573f4e6c287176